rust/kernel/lib.rs

```diff
@@ -138,18 +138,26 @@ impl ThisModule {
 /// Scoped lock on the kernel parameters of [`ThisModule`].
 ///
 /// Lock will be released when this struct is dropped.
+#[repr(transparent)]
 pub struct KParamGuard<'a> {
     this_module: &'a ThisModule,
 }
 
-#[cfg(CONFIG_SYSFS)]
 impl<'a> Drop for KParamGuard<'a> {
+    #[cfg(CONFIG_SYSFS)]
     fn drop(&mut self) {
         // SAFETY: `kernel_param_lock` will check if the pointer is null and
         // use the built-in mutex in that case. The existance of `self`
         // guarantees that the lock is held.
         unsafe { bindings::kernel_param_unlock(self.this_module.0) }
     }
+
+    #[cfg(not(CONFIG_SYSFS))]
+    fn drop(&mut self) {
+        // This just touches the `this_module` field, so the compiler cannot optimize it away,
+        // and so that we don't get a dead_code warning´
```

Comment:
This doesn't change anything about optimizability. An `let _ =` codegens to just evaluating the rhs without storing it everything. In this case the rhs doesn't have side-effects and as such gets optimized away. If LLVM can track all references to `KParamGuard` it can completely remove this field. If not, it will leave it existing.

Reply:
Okay, what should we do than? Also did this to remove the dead code warning

Reply:
It is fine for fixing the dead code warning. The comment just needs to be updated to only mention that it prevnts the dead code warning and not that it prevents optimizations.